core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -191,6 +191,11 @@ class DAGScheduler(
 
   private[scheduler]
   def getCacheLocs(rdd: RDD[_]): Seq[Seq[TaskLocation]] = cacheLocs.synchronized {
+    // Note: if the storage level is NONE, we don't need to get locations from block manager.
+    if (rdd.getStorageLevel == StorageLevel.NONE) {
+      return Seq.fill(rdd.partitions.size)(Nil)
```

Comment:
As a general style note, I'd try to avoid using `return` in Scala code, since there are some corner-cases where using it can lead to exception-handling issues (plus it results in slightly inefficient code which uses exceptions for control flow).